ui/src/app/applications/components/application-sync-panel/application-sync-panel.tsx

```diff
@@ -93,6 +94,17 @@ export const ApplicationSyncPanel = ({application, selectedResource, hide}: {app
                                         return;
                                     }
                                 }
+                                if (prune) {
```

Comment:
I think prompting on _every_ prune is probably excessive. I think we should only prompt if the prune would remove _all_ resources managed by the app.

Reply:
Agree. 
I've modified the code to show the prune msg only when every resource is set to require pruning as true.

```
app status e.g.
 - group: networking.istio.io
    kind: VirtualService
    name: istio-host-split-vsvc
    namespace: argo-rollouts-istio
    requiresPruning: true
    status: OutOfSync
    version: v1beta1

//logic
const requiresPrune = resources.filter(resource => resource?.requiresPruning === true);
[if (prune && appResources.length === requiresPrune.length) {
](https://github.com/argoproj/argo-cd/pull/16603/files#diff-bc9fc4a6bd2a081fba5e3af9aa298ccf3a375d937cf986da719615903f91b3a5R62-R78)```